sklearn/cluster/hierarchical.py

```diff
@@ -609,8 +610,7 @@ class AgglomerativeClustering(BaseEstimator, ClusterMixin):
         "manhattan", "cosine", or 'precomputed'.
         If linkage is "ward", only "euclidean" is accepted.
 
-    memory : Instance of sklearn.externals.joblib.Memory or string, optional \
-            (default=None)
+    memory : Instance of joblib.Memory or string, optional (default=None)
```

Comment:
For type info, I would use joblib.Memory-like everywhere, in this case I would do:
```
memory : joblib.Memory-like, optional (default None)
    # same description as before goes here
```